Utils.java

```diff
@@ -35,5 +40,16 @@ static int rndInRangeWeighted(int[] arr) {
     assert arr.length > 0;
     return arr[rndInRange(0, arr.length - 1)];
   }
+
+  static <K> K getRnd(Map<K, Integer> map) {
```

Comment:
This is functions returns one key map? Maybe, change name this function? getRndKeymap or getRndKey?
No, also using weight, which is the value. From the name of function is not clear what it does.
getRndKeyWithWeight - or very sofisticated?

Reply:
New name sounds good to me, I have updated the code accordingly. Thanks!